### PR TITLE
[NFC] [DebugInfo] Test Value field in SPIRV TypeMember

### PR DESCRIPTION
**Pull request by LU-JOHN**
Test Value field in SPIRV TypeMember.

Also remove some unreachable code.

### CODE REVIEW

File: test/DebugInfo/DebugTypeMember.ll [hunk 1]
```diff
@@ -0,0 +1,63 @@
+; Exercise Value field in SPIRV TypeMember.
```
Comment (MrSidims) [inline test/DebugInfo/DebugTypeMember.ll:1]: is this test in principle different from https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/test/DebugInfo/NonSemantic/static_member_array.ll ?

**Review by MrSidims — COMMENTED** [review]
Removing code LGTM, not sure about the test

File: test/DebugInfo/DebugTypeMember.ll (hunk 1 above)
Comment (LU-JOHN) [reply test/DebugInfo/DebugTypeMember.ll:1]: The unique part of this test is that it has an DIFlagStaticMember and extraData field.  This exercises code in LLVMToSPIRVDbgTran.cpp lines 1016 and 1045.

File: test/DebugInfo/DebugTypeMember.ll [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+; }
+
+; RUN: llvm-as %s -o %t.bc
+
+; RUN: llvm-spirv -o %t.spv %t.bc
+; RUN: llvm-spirv -r -o %t.rev.bc %t.spv
+; RUN: llvm-dis %t.rev.bc -o %t.rev.ll
+; RUN: FileCheck %s --input-file %t.rev.ll
+
+; RUN: llvm-spirv -o %t.ns.spv %t.bc --spirv-debug-info-version=nonsemantic-shader-200
```
Comment (MrSidims) [inline test/DebugInfo/DebugTypeMember.ll:23]: we should probably extend testing with --spirv-debug-info-version=nonsemantic-shader-100
and by adding few SPIR-V checks, other then that - LGTM

**Review by MrSidims — APPROVED** [review]

**Review by MrSidims — APPROVED** [review]

**Pull request merged**